lightning/src/chain/onchaintx.rs

```diff
@@ -388,23 +427,71 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 				assert!(predicted_weight >= transaction.weight());
-				return Some((new_timer, new_feerate, transaction))
+				return Some((new_timer, new_feerate, OnchainClaim::Tx(transaction)))
 			}
 		} else {
-			// Note: Currently, amounts of holder outputs spending witnesses aren't used
-			// as we can't malleate spending package to increase their feerate. This
-			// should change with the remaining anchor output patchset.
-			if let Some(transaction) = cached_request.finalize_package(self, 0, self.destination_script.clone(), logger) {
-				return Some((None, 0, transaction));
+			// Untractable packages cannot have their fees bumped through Replace-By-Fee. Some
+			// packages may support fee bumping through Child-Pays-For-Parent, indicated by those
+			// which require external funding.
+			let inputs = cached_request.inputs();
+			debug_assert_eq!(inputs.len(), 1);
+			let tx = match cached_request.finalize_untractable_package(self, logger) {
+				Some(tx) => tx,
+				None => return None,
+			};
+			if !cached_request.requires_external_funding() {
+				return Some((None, 0, OnchainClaim::Tx(tx)));
 			}
+			return inputs.iter().find_map(|input| match input {
+				// Commitment inputs with anchors support are the only untractable inputs supported
+				// thus far that require external funding.
+				PackageSolvingData::HolderFundingOutput(..) => {
+					debug_assert_eq!(tx.txid(), self.holder_commitment.trust().txid(),
+						"Holder commitment transaction mismatch");
+					// We'll locate an anchor output we can spend within the commitment transaction.
+					let funding_pubkey = &self.channel_transaction_parameters.holder_pubkeys.funding_pubkey;
+					match chan_utils::get_anchor_output(&tx, funding_pubkey) {
+						// An anchor output was found, so we should yield a funding event externally.
+						Some((idx, _)) => {
+							// Our target feerate will depend on whether we have any HTLCs present
+							// within our commitment.
+							let conf_target = if self.holder_commitment.trust().htlcs().is_empty() {
+								ConfirmationTarget::Background
+							} else {
+								ConfirmationTarget::HighPriority
+							};
+							let package_target_feerate_sat_per_1000_weight = cached_request
+								.compute_package_feerate(fee_estimator, conf_target);
+							Some((
+								new_timer,
+								package_target_feerate_sat_per_1000_weight as u64,
+								OnchainClaim::Event(ClaimEvent::BumpCommitment {
+									package_target_feerate_sat_per_1000_weight,
+									commitment_tx: tx.clone(),
+									anchor_output_idx: idx,
+								}),
+							))
+						},
+						// An anchor output was not found. There's nothing we can do other than
+						// attempt to broadcast the transaction with its current fee rate and hope
+						// it confirms. This is essentially the same behavior as a commitment
+						// transaction without anchor outputs.
+						None => Some((None, 0, OnchainClaim::Tx(tx.clone()))),
```

Comment:
IIRC, the case where we don't find a second anchor output for the channel is when we have no fund at stakes in the channel (no-htlc, no-balance) output. We could consider the channel early closed and return a `OnchainEvent::ChannelClosed`, to allow consumer to react in consequence.